yapper/src/main.rs

```diff
@@ -0,0 +1,105 @@
+//! The Yapper emulates user wallet software. The idea is to multicast
+//! a transaction to the entire committee. The yapper... yaps about its
+//! transactions to everyone. The transactions are unstructured bytes, but
+//! they're the *same* unstructured bytes for each node in the committee
+//! due to the requirement of Timeboost.
+
+use std::path::PathBuf;
+
+use anyhow::{Context, Result};
+use cliquenet::Address;
+
+use clap::Parser;
+use timeboost::keyset::{KeysetConfig, wait_for_live_peer};
+use timeboost_utils::types::logging::init_logging;
+
+mod tx;
+
+#[derive(Parser, Debug)]
+struct Cli {
+    /// The number of nodes that are being run in this instance.
+    #[clap(long)]
+    nodes: Option<usize>,
+
+    /// Path to file containing the keyset description.
+    ///
+    /// The file contains backend urls and public key material.
+    #[clap(long)]
+    keyset_file: PathBuf,
+
+    #[clap(long, short, default_value_t = 100)]
+    tps: u32,
+
+    /// Specify how to read the configuration file.
+    #[clap(long, default_value_t = false)]
+    multi_region: bool,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    init_logging();
+
+    let cli = Cli::parse();
+
+    // The total number of nodes that are running.
+    let num = cli.nodes.unwrap_or(4);
+
+    // Unpack the keyset file which has the urls
+    let keyset = KeysetConfig::read_keyset(&cli.keyset_file).context(format!(
+        "opening the keyfile at path {}",
+        cli.keyset_file.to_string_lossy(),
+    ))?;
+
+    // TODO: make this selection a helper.
+    // Rust is *really* picky about mixing iterators, so we just erase the type.
+    let peer_host_iter: Box<dyn Iterator<Item = &_>> = if cli.multi_region {
```

Comment:
I'll smooth this out in a subsequent PR. The offset pattern is copied now in 3 spots.